node/src/config.rs

```diff
@@ -817,8 +822,9 @@ impl DeploymentPlacer for Deployment {
                     .indexers
                     .iter()
                     .map(|idx| {
-                        NodeId::new(idx.clone())
-                            .map_err(|()| format!("{} is not a valid node name", idx))
+                        NodeId::new(idx.clone()).map_err(|err| {
```

Comment:
You can use `.context` here instead of adding `with error {}` to the message.

Reply:
This one I'm kinda confused, once I use `.with_context` I'm getting this error ->   "`?` couldn't convert the error to `std::string::String`: the trait `From<graph::prelude::Error>` is not implemented for `std::string::String`", no entirely sure how to get around this, any thoughts? 🙏🏽

Reply:
I didn't understand either, not the prettiest thing, but here's the solution:
```rust
NodeId::new(idx.clone())
    .context(|| format!("{} is not a valid node name", idx))
    .map_err(|e| e.to_string())
```

Reply:
Thanks, the only way I managed to fix this is like so:

```
      .map(|idx| {
          NodeId::new(idx.clone())
              .with_context(|| format!("{:#} is not a valid node name", idx))
      })
      .collect::<Result<Vec<_>, _>>()
      .unwrap();

```

My understanding is that since we don't return a Result (to deal w/  possible errors) using unwrap would not be a bad idea (maybe?!) 😬  but again not sure about that also.

Which solution you think is more appropriate for this situation? I'm just suggesting mine because it required a little bit less  coe